File: 136. Single Number/singleNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    let result = nums[0]
    for (i=1;i<nums.length;i++) {
        result ^= nums[i] // XOR
    }
    return result
};

// Runtime 89 ms / Memory 43.1 MB
// debug: https://jsfiddle.net/

File: 136. Single Number/singleNumber.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int r = nums[0];
        for (int i=1; i<nums.size(); i++) {
            r = r ^ nums[i]; //XOR:00->0 11->0 01->1
        }
        return r;
    }
};